client/state/document-head/reducer.js

```diff
@@ -18,7 +18,7 @@ import {
 export function title( state = '', action ) {
 	switch ( action.type ) {
 		case DOCUMENT_HEAD_TITLE_SET:
-			return action.title;
+			return action.title || state;
```

Comment:
I guess this might mean I need to guard agains `undefined` title in `<DocumentHead/>`'s `componentWillReceiveProps()` after all ([related](https://github.com/Automattic/wp-calypso/pull/7224#discussion_r74437843)).